src/Handler/MockHandler.php

```diff
@@ -87,6 +87,19 @@ function ($value) use ($request, $options) {
                 if ($this->onFulfilled) {
                     call_user_func($this->onFulfilled, $value);
                 }
+                if (isset($options['sink'])) {
+                    $contents = (string) $value->getBody();
+                    $sink = $options['sink'];
+
+                    if (is_resource($sink)) {
```

Comment:
I think this could be cleaned up and made more memory efficient for files and stream instances by updating this to something like:

```php
$sink = $options['sink'];

if (is_string($sink)) {
    file_put_contents($sink, $value->getBody());
} else {
    $contents = Psr7\stream_for($sink);
    Psr7\copy_to_stream($body->getContents(), $sink);
}
```